src/packet.rs

```diff
@@ -352,7 +358,59 @@ pub mod play {
         0x23 => BlockChange { location: BlockPos, block_id: Var<i32> }
         0x24 => BlockAction { location: BlockPos, byte1: u8, byte2: u8, block_type: Var<i32> }
         0x25 => BlockBreakAnimation { entity_id: Var<i32>, location: BlockPos, destroy_stage: i8 }
-        // 0x26 => MapChunkBulk { sky_light_sent: bool, chunks: Vec<Chunk>; impl Protocol for MapChunkBulk { ... } } // PROBLEM: chunks is encoded as two arrays, the first one specifying which sections of each chunk column are empty
+        0x26 => ChunkDataBulk { sky_light_sent: bool, chunk_meta: Vec<ChunkMeta>, chunk_data: Vec<ChunkColumn>;
+            impl Protocol for ChunkDataBulk {
+                type Clean = Self;
+                fn proto_len(this: &Self) -> usize {
+                    let mut len = 1; // sky_light_sent(bool) len is constant
```

Comment:
```rust
fn proto_len(this: &Self) -> usize {
    let columns = this.chunk_meta.len() as i32;
    1 // sky_light_sent(bool) len is constant
    + <Var<i32> as Protocol>::proto_len(&columns)
    + this.chunk_meta.iter().map(|cm| <ChunkMeta as Protocol>::proto_len(cm)).sum()
    + this.chunk_data.iter().map(|cd| cd.len()).sum()
}
```

Requires the [`AdditiveIterator`](http://doc.rust-lang.org/std/iter/trait.AdditiveIterator.html) trait.